lampod-cli/src/main.rs

```diff
@@ -44,21 +48,61 @@ fn main() -> error::Result<()> {
     Ok(())
 }
 
+fn write_words_to_file<P: AsRef<Path>>(path: P, words: Option<String>) -> io::Result<()> {
+    let mut file = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(path)?;
+
+    match words {
+        Some(ref value) => {
+            file.write_all(value.as_bytes())?;
+        }
+        None => {}
+    }
+
+    Ok(())
+}
+
+fn load_words_from_file<P: AsRef<Path>>(path: P) -> io::Result<Option<String>> {
+    let mut file = File::open(path)?;
+    let mut content = String::new();
+
+    file.read_to_string(&mut content)?;
+
+    if content.is_empty() {
+        Ok(None)
+    } else {
+        Ok(Some(content))
+    }
+}
+
 /// Return the root directory.
 fn run(args: LampoCliArgs) -> error::Result<()> {
-    let mnemonic = if args.restore_wallet {
-        let inputs: String = term::input(
-            "BIP 39 Mnemonic",
-            None,
-            Some("To restore the wallet, lampo needs a BIP39 mnemonic with words separated by spaces."),
-        )?;
-        Some(inputs)
+    let restore_wallet = args.restore_wallet;
+
+    // After this point the configuration is ready!
+    let mut lampo_conf: LampoConf = args.try_into()?;
+
+    let words_path = format!("{}/", lampo_conf.path());
+    let mnemonic = if restore_wallet {
+        if Path::new(&format!("{}/wallet.dat", words_path)).exists() {
+            // Load the mnemonic from the file
+            load_words_from_file(format!("{}/wallet.dat", words_path))?
+        } else {
+            // If file doesn't exist, ask for user input
+            let inputs: String = term::input(
+                "BIP 39 Mnemonic",
+                None,
+                Some("To restore the wallet, lampo needs the BIP39 mnemonic with words separated by spaces."),
+            )?;
+            Some(inputs)
+        }
```

Comment:
```suggestion
        if Path::new(&format!("{}/wallet.dat", words_path)).exists() {
            // Load the mnemonic from the file
            load_words_from_file(format!("{}/wallet.dat", words_path))?
        } else {
            // If file doesn't exist, ask for user input
            let inputs: String = term::input(
                "BIP 39 Mnemonic",
                None,
                Some("To restore the wallet, lampo needs the BIP39 mnemonic with words separated by spaces."),
            )?;
            inputs
        }
```

What is the reason to exit from this if with a Some? you know that will be always some. So at this point the mnemonic should be known.

Then in case we are not restoring the wallet, we need to do two things:

1. The file exist? if yes return an error to the user, otherwise ldk will messup with the node secrets and you are losing funds
2. if the wallet file do not exist, generate the mnemonic. See a couple of line down `CoreWalletManager::new(Arc::new(lampo_conf.clone()))?`  probably it is time to refactor this code

Reply:
>The advantage of using some here is that it makes mnemonic an Option type. We need it to be option because we also have a none case, the case in which restore wallet is not called.

The mnemonic to generate the wallet must be always define, I do not understand where the None case is helping you?

Reply:
The none case is later checked to generate new words here. https://github.com/Psycho-Pirate/lampo.rs/blob/fbdf05798fad7ab7a264cb33c04398a847bca55d/lampod-cli/src/main.rs#L146

If mnemonic is set to none, it means that we are having a new lampo session. Hence new words are generated.

Reply:
Naa you are overcomplicating it or trusting too much the lampo shitty code

What do you think about https://github.com/vincenzopalazzo/lampo.rs/pull/295#issuecomment-2453061580 ?